gw_eccentricity/eccDefinition.py

```diff
@@ -1184,6 +1228,180 @@ def interp_extrema(self, extrema_type="pericenters"):
+                self.rational_fit_kwargs["num_degree"], self.rational_fit_kwargs["denom_degree"] \
+                    = new_num_degree, new_denom_degree
+                new_rat_fit = self.get_rat_fit(x, y)
+                if self.check_if_domega_dt_is_nonmonotonic(t, new_rat_fit(t), 1.0, description):
+                    # Revert to previous fit and degrees if nonmonotonicity is detected
+                    debug_message(f"Increasing degrees caused nonmonotonicity. Reverting to "
+                                  f"last monotonic fit with num_degree={last_monotonic_num_degree} "
+                                  f"and denom_degree={last_monotonic_denom_degree}.",
+                                  debug_level=self.debug_level, important=False)
+                    self.rational_fit_kwargs["num_degree"] = last_monotonic_num_degree
+                    self.rational_fit_kwargs["denom_degree"] = last_monotonic_denom_degree
+                    rat_fit = last_monotonic_rat_fit
+                    break
+
+                last_monotonic_rat_fit = new_rat_fit
+                last_monotonic_num_degree = new_num_degree
+                last_monotonic_denom_degree = new_denom_degree
+                rat_fit = new_rat_fit
+
+        # update final degrees used to build the fit
+        self.rational_fit_degrees[description] = (self.rational_fit_kwargs["num_degree"],
+                                                  self.rational_fit_kwargs["denom_degree"])
+        return rat_fit
+
+    def get_approximate_degree_for_rational_fit(self):
+        """Get approximate degree based on the number of extrema.
+
+        Assign degree based on number of extrema found. The degree is increased as
+        the number of extrema increases.
+        """
+        # TODO: Optimize this.
+        # assign degree based on the number of extrema if user provided
+        # degree is None.
+        approximate_num_orbits = max(len(self.pericenters_location),
+                                     len(self.apocenters_location))
+        if approximate_num_orbits <= 5:
+            num_degree, denom_degree = 1, 1
+        elif (approximate_num_orbits > 5) and (approximate_num_orbits <= 20):
+            num_degree, denom_degree =  2, 2
+        elif (approximate_num_orbits > 20) and (approximate_num_orbits <= 50):
+            num_degree, denom_degree = 3, 3
+        elif (approximate_num_orbits > 50) and (approximate_num_orbits <= 100):
+            num_degree, denom_degree = 4, 4
+        elif (approximate_num_orbits > 100) and (approximate_num_orbits <= 150):
+            num_degree, denom_degree = 5, 5
+        elif (approximate_num_orbits > 150) and (approximate_num_orbits <= 200):
+            num_degree, denom_degree = 6, 6
+        else:
+            num_degree, denom_degree = 5 + int(np.log10(approximate_num_orbits)), 5 + int(np.log10(approximate_num_orbits))
+        return num_degree, denom_degree
```

Comment:
If you check for approximate_num_orbits <= 5 once, you don't need to check for approximate_num_orbits > 5 again in elif. Something like:
```
if orbits <= 5:
    bla
elif orbits <= 20:
```

Reply:
How did you arrive at these choices? Add some comments about how you got there and how important these particular choices are.

Reply:
corrected. These are based on limited investigations. It needs a lot more tests to set more optimal values which I intend to do once this is merged. I have noted this using TODO at the top